subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -126,8 +126,13 @@ static int getaddrinfo_null_host(int port, const struct zsock_addrinfo *hints,
 		addr->sin_family = AF_INET;
 		INIT_ADDRINFO(res, addr);
 		res->ai_family = AF_INET;
-		res->ai_socktype = SOCK_STREAM;
-		res->ai_protocol = IPPROTO_TCP;
+		if (hints->ai_socktype == SOCK_DGRAM) {
+			res->ai_socktype = SOCK_DGRAM
```

Comment:
we are missing `;` here

Reply:
Thanks, I'm currently fixing this, and adding tests for the whole null_host resolver.